tests/components/tedee/test_init.py

```diff
@@ -67,3 +67,38 @@ async def test_bridge_device(
     )
     assert device
     assert device == snapshot
+
+
+async def test_cleanup_disconnected_locks(
+    hass: HomeAssistant,
+    mock_config_entry: MockConfigEntry,
+    mock_tedee: MagicMock,
+    device_registry: dr.DeviceRegistry,
+    entity_registry: er.EntityRegistry,
+    snapshot: SnapshotAssertion,
+) -> None:
+    """Ensure disconnected locks are cleaned up."""
+    mock_config_entry.add_to_hass(hass)
+    await hass.config_entries.async_setup(mock_config_entry.entry_id)
+    await hass.async_block_till_done()
+
+    def assert_current_registry_state() -> None:
+        devices = dr.async_entries_for_config_entry(
+            device_registry, mock_config_entry.entry_id
+        )
+        assert devices == snapshot
+
+        for device in devices:
+            entities = er.async_entries_for_device(
+                entity_registry, device.id, include_disabled_entities=True
+            )
+            assert entities == snapshot
```

Comment:
Idem, when the device is removed we are now testing if two entities aren't in the list.